pkg/rtc/transport.go

```diff
@@ -924,6 +922,10 @@ func (t *PCTransport) SendDataPacket(dp *livekit.DataPacket, data []byte) error
 		return ErrDataChannelUnavailable
 	}
 
+	if t.pc.ConnectionState() == webrtc.PeerConnectionStateFailed {
+		return ErrTransportFailure
+	}
```

Comment:
Adding this here as there were attempts to send data packet after ICE/peer connection failure.